sdk/src/azure/core/az_json_writer.c

```diff
@@ -829,26 +825,18 @@ static AZ_NODISCARD az_result _az_json_writer_append_literal(
     remaining_json = az_span_copy_u8(remaining_json, ',');
   }
 
-  remaining_json = az_span_copy(remaining_json, literal);
+  az_span_copy(remaining_json, literal);
```

Comment:
I think all non-void functions should be AZ_NODISCARD.
Then if value is return value is not used again and that's a warning, we should explicitly add `void(variable)` so it is clear about the ignore for the variable.
No need to change it, however, it's really not too much trouble

Reply:
> I think all non-void functions should be AZ_NODISCARD.

I am not sure I agree with doing this for _all_.

For example, in this case, it is totally reasonable for the user to not care about the remaining slice, especially after a call to a null terminating az_span_copy_u8. We shouldn't force the caller to jump through hoops when not necessary.

Reply:
isn't the reason of asking the NODISCARD that the function wants to make sure that it is the user the one who might ignore the result but never the function itself?

I agree that it is totally reasonable for the user to not care about the remaining slice, but I believe it is the user who should take the call by getting getting the result and explicitly ignoring it.

I'm saying that all non-void functions in the SDK should follow the same pattern..  either letting user implicitly ignore the function return or always doing the NODISCARD.

But I don't really care that much at the end